internal/controller/pipelinerollout_controller.go

```diff
@@ -270,14 +272,32 @@ func processPipelineStatus(ctx context.Context, pipeline *kubernetes.GenericObje
 
 	numaLogger.Debugf("pipeline status: %+v", pipelineStatus)
 
+	// TODO: make string comparison instead of using numaflowv1 import (it defeats the purpose of having RawExtension).
+	// Create a function to convert/parse/check string instead of numaflowv1.Phase
+	// Do this for all 3 controllers
+
 	pipelinePhase := numaflowv1.PipelinePhase(pipelineStatus.Phase)
 	switch pipelinePhase {
 	case numaflowv1.PipelinePhaseFailed:
-		pipelineRollout.Status.MarkChildResourcesUnhealthy("PipelineFailed", "Pipeline Failed")
+		pipelineRollout.Status.MarkChildResourcesUnhealthy("PipelineFailed", "Pipeline Failed", pipelineRollout.Generation)
+	case numaflowv1.PipelinePhasePaused, numaflowv1.PipelinePhasePausing:
+		// TODO: update string(pipelinePhase) when working on strng comparison changes
+		pipelineRollout.Status.MarkPipelinePausingOrPausedWithReason(string(pipelinePhase), pipelineRollout.Generation)
+		if pipeline.Generation == pipelineStatus.ObservedGeneration {
+			pipelineRollout.Status.MarkChildResourcesHealthy(pipelineRollout.Generation)
```

Comment:
So, it looks like if it's paused or pausing, but the generation of the child matches its observedGeneration, we mark it healthy. Personally, I prefer that. But want to verify @xdevxy is okay with that, since it seems like her document proposed it would be unhealthy, right?

Reply:
I don't feel strong about this as long as we have a condition to indicate paused/pausing specifically. However if we go with this logic, this needs to be updated https://github.com/numaproj/numaplane/pull/92/files/7b887a13f05abde419720205788e4d325d65f39a#diff-e8e263da81dda7f4a11f5070b4e4c2571f119bbd43ac576a8d32f3af098d7ae7R44-R45 accordingly. My doc was just referring to this.

Reply:
I will update that comment